docs/guide/snaps-rpc-api.md

```diff
@@ -447,6 +447,166 @@ if (result === true) {
 }
 ```
 
+### `snap_dialog`
+
+::: warning Only Callable By
+
+- Snaps
+  :::
+
+#### Description
+
+Calling this method causes a dialog to be displayed in the MetaMask UI.
+There are three types of dialogs: Alert, Confirmation, and Prompt.
+Each of these dialog types has different parameters and return types, detailed below.
+
+#### The `Component` type
```

Comment:
@GuillaumeRx this is my attempt at explaining the `Component` type. Looking forward to your comments on it.

Reply:
@ziad-saab lot's of UI types are inspired by [unist](https://github.com/syntax-tree/unist) Abstract Syntax Trees, might be useful for inspiration

Reply:
@ritave How would that help the common Snaps developer to understand what the `Component` type is? I don't think my explanation is great, but in reality it's an internal type so I'm not sure how to convey that. It sounds like `unist` is an implementation detail, I don't see how to incorporate it into a satisfactory explanation of what `Component` is. Any ideas?